Foundation/FileManager.swift

```diff
@@ -2377,46 +2377,107 @@ extension FileManager {
             self.innerEnumerator = ie
         }
         
-        @available(Windows, deprecated, message: "Not Yet Implemented")
         override func nextObject() -> Any? {
             let o = innerEnumerator.nextObject()
             guard let url = o as? URL else {
                 return nil
             }
 
 #if os(Windows)
-            NSUnimplemented()
+            let path = url.path.replacingOccurrences(of: baseURL.path+"\\", with: "")
 #else
             let path = url.path.replacingOccurrences(of: baseURL.path+"/", with: "")
+#endif
             _currentItemPath = path
             return path
-#endif
         }
     }
 
-#if os(Windows)
     internal class NSURLDirectoryEnumerator : DirectoryEnumerator {
-        internal typealias ErrorHandler = /* @escaping */ (URL, Error) -> Bool
+#if os(Windows)
+        var _options : FileManager.DirectoryEnumerationOptions
+        var _errorHandler : ((URL, Error) -> Bool)?
+        var _stack: [URL]
+        var _current: URL?
+        var _root_depth : Int
+
 
-        init(url: URL, options: FileManager.DirectoryEnumerationOptions, errorHandler: ErrorHandler?) {
+        init(url: URL, options: FileManager.DirectoryEnumerationOptions, errorHandler: (/* @escaping */ (URL, Error) -> Bool)?) {
+            _options = options
+            _errorHandler = errorHandler
+            _stack = [url]
+            _root_depth = url.pathComponents.count
+        }
+
+
+        override func nextObject() -> Any? {
+            func contentsOfDir(directory: URL) -> [URL]? {
+                var ffd: WIN32_FIND_DATAW = WIN32_FIND_DATAW()
+                let dirPath = Array((directory.absoluteString + "\\*\0").utf16)
+                guard let h = dirPath.withContiguousStorageIfAvailable(
+                    {FindFirstFileW($0.baseAddress, &ffd)}),
```

Comment:
The lack of spaces around the closure make this hard to comprehend.  Remove the parenthesis:

```
let h: HANDLE = dirPath.withContiguousStorageIfAvailable {
  FindFirstFileW($0.baseAddress, &ffd)
}
guard h != INVALID_HANDLE_VALUE else { return nil }
```

Is easier to read IMO.